components/screens/MainScreen.tsx

```diff
@@ -82,23 +108,86 @@ class MainScreen extends React.Component<Props, State> {
    * Component render
    */
   public render() {
-      return (
-        <BasicLayout backgroundColor="#fff">
-          <Text>Course Overview</Text>
-        </BasicLayout>
-      );
-    }
+    return (
+      <BasicLayout loading={this.state.loading} backgroundColor="#fff">
+        <FlatList
+        style={defaultStyles.listContainer}
+        data={this.state.courseContent}
+        renderItem={({item}) =>
+        <TouchableOpacity onPress= {() => this.onTopicPress(item)}>
+          <View style={[defaultStyles.topicItemBase,
+            item.topicDone ? styles.itemDone : (item.topicAvailable ? defaultStyles.topicItemBase : styles.topicItemInactive)]}>
+            <Icon containerStyle={defaultStyles.taskIcon} size={46} name={item.topicDone ? "trophy" : "eye"} type="evilicon" color="white"/>
+            <View style={defaultStyles.topicTaskIconBackground}/>
+            <Text style={[defaultStyles.topicItemText, item.topicDone ? defaultStyles.topicItemText : styles.itemActiveText]}>{item.topicName}</Text>
+            <Icon containerStyle={defaultStyles.progressIcon} size={50} name={item.topicDone ? "check" : "arrow-right"} type="evilicon"/>
+          </View>
+        </TouchableOpacity>}
+        keyExtractor={(item, index) => index.toString()}
+        />
+      </BasicLayout>
+    );
+  }
 
   /**
-   * Returns topics from moodle Api
+   * Returns the courses topics from moodle Api
    */
-  private async getTopicsFromMoodle() {
-    if (!this.state.moodleToken) {
+  private async getTopicsFromMoodle(courseId: number) {
+    if (!this.props.moodleToken) {
       return this.props.navigation.navigate("Login");
     }
 
-    // TODO Get course topics from moodle and display them
-    const moodleService = Api.getMoodleService("https://ppo-test.metatavu.io", this.state.moodleToken);
+    const moodleService = await Api.getMoodleService("https://ppo-test.metatavu.io", this.props.moodleToken);
+
+    const topics: any = await moodleService.coreCourseGetContents({courseid: courseId});
+
+    const quizService = await Api.getModQuizService("https://ppo-test.metatavu.io", this.props.moodleToken);
+
+    const quizList: any = await quizService.getQuizzesByCourses({courseids: [courseId]});
+
+    const courseContent: CourseTopic[] = [];
+
+    for (const topic of topics) {
+      const newCourseItem: CourseTopic = {
+        id: topic.id,
+        topicName: topic.name,
+        topicDone: true,
+        topicAvailable: true,
+        topicContent: []
+      }
+      if (topic.modules.length === 0) {
+        newCourseItem.topicDone = false
+      }
+
+      for (const activity of topic.modules) {
```

Comment:
const?

Reply:
const can be used in for of loops if you don't reassign the variable. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of